src/v3/src/components/Widget/index.tsx

```diff
@@ -84,16 +85,41 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
       return createForm();
     }
 
-    if (TERMINAL_STATUSES.includes(idxTransaction.status) || !idxTransaction.nextStep) {
+    if ([IdxStatus.TERMINAL, IdxStatus.SUCCESS].includes(idxTransaction.status)
+        || !idxTransaction.nextStep) {
       return transformTerminalTransaction(idxTransaction, widgetProps);
     }
 
-    const bag = transformTransaction(idxTransaction as IdxTransactionWithNextStep, widgetProps);
+    let step = stepToRender;
+    // Mobile devices cannot scan QR codes while navigating through flow
+    // so we force them to select either email / sms for enrollment
+    if (idxTransaction.context.currentAuthenticator?.value.key === AuthenticatorKey.OKTA_VERIFY
+        && idxTransaction.nextStep.name === 'enroll-poll'
+        && isAndroidOrIOS()) {
+      step = 'select-enrollment-channel';
+    }
```

Comment:
As I mentioned before, all logic like this are side effects, and they should be managed out of transformation layer, here I didn't do too much refactor to take all side effects out.

Eventually, server should know the device info based on requests, and serve remediations accordingly, so keep these logic out of transformation layer can also make future changes easier.

Reply:
Can a function be created out of the condition so its clear and concise as to what is the purpose of this to future Engs?

Reply:
possible, but not prefer personally, since assigning a good name to that util is not very concise, and where to put it becomes another question. I would prefer to organize logic like these when a pattern has been discovered.